File: src/main/java/hilfling/backend/hilfing/model/PhotoGangBangerPosition.java
package hilfling.backend.hilfing.model;

import lombok.Data;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.io.Serializable;

@Data
@Entity(name = "photo_gang_banger_positions")
@Table(name = "photo_gang_banger_positions"  , uniqueConstraints = {
        @UniqueConstraint(columnNames = {"photo_gang_banger_id", "position_id"})
})
public class PhotoGangBangerPosition implements Serializable, BaseModel {
    public PhotoGangBangerPosition() {
    }

    public PhotoGangBangerPosition(PhotoGangBanger photoGangBanger, Position position, Boolean current) {
        this.photoGangBanger = photoGangBanger;
        this.position = position;
        this.current = current;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "photo_gang_banger_id", referencedColumnName = "id", nullable = false)
    private PhotoGangBanger photoGangBanger;

    @ManyToOne
    @JoinColumn(name = "position_id", referencedColumnName = "id", nullable = false)
    private Position position;

    @Column(name = "current", nullable = false)
    @ColumnDefault("true")
    private Boolean current;
}

File: src/main/kotlin/hilfling/backend/hilfling/model/PhotoGangBangerPosition.kt
package hilfling.backend.hilfing.model

import hilfling.backend.hilfing.model.Photos.references
import me.liuwj.ktorm.database.Database
import me.liuwj.ktorm.entity.Entity
import me.liuwj.ktorm.entity.sequenceOf
import me.liuwj.ktorm.schema.*

interface PhotoGangBangerPosition : BaseModel<PhotoGangBangerPosition> {
    companion object : Entity.Factory<PhotoGangBangerPosition>()

    var isCurrent: Boolean

    // Foreign keys
    var position: Position
    var photoGangBanger: PhotoGangBanger
}

object PhotoGangBangerPositions : BaseTable<PhotoGangBangerPosition>("photo_gang_banger_position") {
    val isCurrent = boolean("is_current").bindTo { it.isCurrent }

    // Foreign keys
    val position = int("position_id").references(Positions){it.position}
    val photoGangBanger = int("photo_gang_banger_id").references(PhotoGangBangers){it.photoGangBanger}

}

val Database.photo_gang_banger_positions get() = this.sequenceOf(PhotoGangBangerPositions)